server/datastore/mysql/migrations/tables/20240905140515_AddPoliciesToNoTeam.go

```diff
@@ -0,0 +1,78 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240905140515, Down_20240905140515)
+}
+
+func Up_20240905140515(tx *sql.Tx) error {
+	//
+	// Changes in `policies` and `policy_stats` to support policies for "No team".
+	// "No team" here means policies that run on hosts that belong to no team (hosts.team_id = NULL)
+	//
+	// `policies`:
+	//	- team_id = NULL means the policy is a "Global policy" (aka "All teams" policy).
+	//	- team_id > 0 means the policy is a team policy.
+	//	- team_id = 0 means the policy is a "No team" policy.
+	//
+	// `policy_stats`:
+	// 	- For "Global policies":
+	//	  - inherited_team_id_char = 'global', inherited_team_id = NULL are the stats for the policy's global domain.
+	//	  - inherited_team_id_char = '<TEAM_ID>', inherited_team_id = <TEAM_ID> are the stats of the policy on a specific team domain.
+	//	  - inherited_team_id_car = '0', inherited_team_id = 0 are the stats of the policy on the "No team" domain.
+	//	- For "Team policies" (for team policies there's always just one row in this table):
+	//	  - inherited_team_id_char = 'global', inherited_team_id = NULL are the stats for the team policy.
+	//
+
+	// Drop foreign key on policies table to teams to allow for team_id = 0 to represent "No team".
+	referencedTables := map[string]struct{}{"teams": {}}
+	table := "policies"
+	constraints, err := constraintsForTable(tx, table, referencedTables)
+	if err != nil {
+		return err
+	}
+	if len(constraints) != 1 {
+		return errors.New("policies foreign key to teams not found")
```

Comment:
Why the error here? If there are no constraints, then we don't need to drop the FK, I think.

Reply:
We are assuming the fk is there on all deployments.